app/src/main/java/com/infomaniak/mail/views/AvatarView.kt

```diff
@@ -60,14 +61,20 @@ class AvatarView @JvmOverloads constructor(
     private var currentCorrespondent: Correspondent? = null
     private var currentBimi: Bimi? = null
 
-    private val mergedContactObserver = Observer<MergedContactDictionary> { contacts ->
-        val displayType = getAvatarDisplayType(currentCorrespondent, currentBimi)
+    // We use waitInitMediator over MediatorLiveData because we know both live data will be initialized very quickly anyway
+    private val avatarMediatorLiveData = Utils.waitInitMediator(
+        avatarMergedContactData.mergedContactLiveData,
+        avatarMergedContactData.isBimiEnabledLiveData,
+    )
 
-        if (displayType == AvatarDisplayType.CUSTOM_AVATAR || displayType == AvatarDisplayType.INITIALS) {
-            loadAvatarUsingDictionary(currentCorrespondent!!, contacts)
-        }
+    private val mediatorObserver = Observer<Pair<MergedContactDictionary, Boolean>> { (contacts, isBimiEnabled) ->
```

Comment:
maybe the observer could be named something along the lines of `avatarUpdateObserver`

Reply:
`mediatorObserver` name doesn't tell us anything about what it's about.